data/transactions/logic/eval.go

```diff
@@ -3154,3 +3162,31 @@ func opAppParamsGet(cx *evalContext) {
 	cx.stack[last] = value
 	cx.stack = append(cx.stack, stackValue{Uint: exist})
 }
+
+func opLog(cx *evalContext) {
+	last := len(cx.stack) - 1
+
+	if cx.stack[last].argType() != StackBytes {
+		cx.err = fmt.Errorf("log arg 0 wanted type []byte got uint64")
+		return
+	}
```

Comment:
it is checked automatically in `step()` function but please double check before removal this branch